doc/conf.py

```diff
@@ -47,12 +47,23 @@
     "add_toctree_functions",
     "sphinx-prompt",
     "sphinxext.opengraph",
+    "matplotlib.sphinxext.plot_directive",
 ]
 
 # this is needed for some reason...
 # see https://github.com/numpy/numpydoc/issues/69
 numpydoc_class_members_toctree = False
 
+# Produce `plot::` directives for examples that contain `import matplotlib` or
+# `from matplotlib import`.
+numpydoc_use_plots = True
+
+# Options for the `::plot` directive:
+# https://matplotlib.org/stable/api/sphinxext_plot_directive_api.html
+plot_formats = ["png"]
```

Comment:
`plot_formats` defaults to `['png', 'hires.png', 'pdf']`, just creating a png should be enough:


https://github.com/matplotlib/matplotlib/blob/a8977fde9ae67fc736b29c009eb40eb985924301/lib/matplotlib/sphinxext/plot_directive.py#L108